### DISCUSSION TITLE
Welcome to memory.dll Discussions!

### DISCUSSION THREAD

**Discussion by erfg12** [erfg12/memory.dll#104] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by 7x-hex-x7** [comment]
Hello! How can I implement this code in C#? I want to find a static address. How can i get this address with Memory.dll AoBScan?
`[ENABLE]
aobscan(aob,8B 35 xx xx xx xx 57 8B F8 83 C0 01 83 E0)
label(ptrBase)
registersymbol(ptrBase)
[aob+2]:
ptrBase:
[DISABLE]
unregistersymbol(ptrBase)`
In the disassembler it turns out like this: `Dead Space.exe+5E79C - 8B 35 8C57F400        - mov esi,["Dead Space.exe"+B4578C] { (22160010) }`
But I cannot figure out why one letter is different if it is an address: bytes 8C57**F4**00 and opcode "Dead Space.exe"+00**B4**578C
![Screenshot 2021-3-16_21-30-35 875](https://user-images.githubusercontent.com/41704734/111362091-74a7ad00-869f-11eb-800f-76560d521067.jpg)

**Reply by erfg12** [reply]
First, follow the getting started wiki page: https://github.com/erfg12/memory.dll/wiki/How-To-Get-Started-(C%23)

Then, read the AoB Scan method provided: https://github.com/erfg12/memory.dll/wiki/AoB-Scanning

For your last question, I don't understand what you're asking. If you're asking why `8C57F400` and `8b 35 8C57F400` are in different spots, that's because there are 2 bytes prior to that number, so you have to go back a few hex digits to get it. If I'm wrong, please clarify and add more information to your question.

**Reply by 7x-hex-x7** [reply]
Thank you, I know how to write trainers in C # and I can use AoBScan, I have already made several trainers. But I cannot figure out how to implement this script in C#. That is, look, I am doing AoBScan by bytes: 8B 35 xx xx xx xx 57 8B F8 83 C0 01 83 E0, then I need to somehow get the base pointer (in this case "Dead Space.exe"+00B4578C), as in the screenshot, but I can't figure out how to do it, in the CE script this is done automatically. If I just do AoBScan, then another address will come out, for example, Dead Space.exe+5E79C. How can I get this address out of opcodes, I hope you understand me :)

**Reply by 7x-hex-x7** [reply]
https://guidedhacking.com/threads/c-want-to-read-all-addresses-from-instruction-opcode.12891/post-75236
Here a person has a similar situation, he, like me, needs to read the address from the opcode / instruction, but I have a green static address in the instruction / opcode

**Comment by jjcho849** [comment]
my visual studio won't install the memory.dll

Could not install package 'Memory.dll.x86 1.2.26'. You are trying to install this package into a project that targets '.NETFramework,Version=v4.7.2', but the package does not contain any assembly references or content files that are compatible with that framework. For more information, contact the package author.	

PLEASE HELP!

**Comment by flexeykinDev** [comment]
How can i change opcode values inline, or load a full auto assembler script from cheat engine
```
[ENABLE]
aobscanmodule(bullets,$process,89 45 5C 8B C6)
alloc(newmem,$1000)
label(code return)

newmem:
label(bullet_type)
registersymbol(bullet_type)

  bullet_type:
    dd 00

  code:
    push eax
    cmp [bullet_type],C //12
    jg check12
    cmp [bullet_type],0
    jl check0

inject: //originalcode
  mov eax,[bullet_type]
  mov [ebp+5C],eax
  pop eax
  mov eax,esi
  jmp return

  check0:
    mov [bullet_type],00
    mov eax,[bullet_type]
    jmp inject

  check12:
    mov [bullet_type],#12
    mov eax,[bullet_type]
    jmp inject

bullets:
  jmp code

return:
registersymbol(bullets)

[DISABLE]
bullets:
  db 89 45 5C 8B C6

unregistersymbol(bullets)
dealloc(newmem)
```
for example this code for PvZ can change bullet type. I understand how to makes a code who do nothing is 0x90 and how to change value with pointers, but idk how to change opcodes for example je for jmp and etc.
